src/store/messages/saga.ts

```diff
@@ -133,6 +134,13 @@ export function* fetchNewMessages(action) {
   );
 }
 
+export function* deleteMessage(action) {
+  const { channelId, messageId } = action.payload;
+
+  yield call(deleteMessageApi, channelId, messageId);
+  yield call(fetch, { payload: { channelId } });
+}
```

Comment:
Is it possible to  remove the message from the store before we call the delete API? I think it would be a better user experience.

Does something like this work?
```
export function* deleteMessage(action) {
  const { channelId, messageId } = action.payload;

  const existingMessages = yield select(rawMessagesSelector(channelId));

  yield put(
    receive({
      id: channelId,
      messages: existingMessages.filter(id => { return id !== messageId }),
    })
  );

  yield call(deleteMessageApi, channelId, messageId);
}
```

Reply:
The test could be: 
```   
    const channelId = '280251425_833da2e2748a78a747786a9de295dd0c339a2d95';
    const messages = [
      { id: 1, message: 'This is my first message' },
      { id: 2, message: 'I will delete this message' },
      { id: 3, message: 'This is my third message' },
    ];

    const messageIdToDelete = messages[1].id;

    const initialState = {
      normalized: {
        channels: {
          [channelId]: {
            id: channelId,
            messages: messages.map(m => m.id),
          },
        },
        messages,
      },
    };

    const {
      storeState: {
        normalized
      },
    } = await expectSaga(deleteMessage, { payload: { channelId, messageId: messageIdToDelete } })
      .withReducer(rootReducer, initialState as any)
      .provide([
        [
          matchers.call.fn(deleteMessageApi),
          200,
        ],
      ])
      .run();

    expect(normalized.channels[channelId].messages).toEqual([messages[0].id, messages[2].id]);
```

Reply:
The test could be:
```
   const channelId = '280251425_833da2e2748a78a747786a9de295dd0c339a2d95';
    const messages = [
      { id: 1, message: 'This is my first message' },
      { id: 2, message: 'I will delete this message' },
      { id: 3, message: 'This is my third message' },
    ];

    const messageIdToDelete = messages[1].id;

    const initialState = {
      normalized: {
        channels: {
          [channelId]: {
            id: channelId,
            messages: messages.map(m => m.id),
          },
        },
        messages,
      },
    };

    const {
      storeState: {
        normalized
      },
    } = await expectSaga(deleteMessage, { payload: { channelId, messageId: messageIdToDelete } })
      .withReducer(rootReducer, initialState as any)
      .provide([
        [
          matchers.call.fn(deleteMessageApi),
          200,
        ],
      ])
      .run();

    expect(normalized.channels[channelId].messages).toEqual([messages[0].id, messages[2].id]);
```

Reply:
Yes, it's better like this